ngraph/core/include/ngraph/op/batch_norm.hpp

```diff
@@ -66,6 +65,44 @@ namespace ngraph
                 double m_epsilon;
             };
         } // namespace v0
-        using v0::BatchNormInference;
+        namespace v5
+        {
+            class NGRAPH_API BatchNormInference : public Op
+            {
+            public:
+                NGRAPH_RTTI_DECLARATION;
+                BatchNormInference() = default;
+                /// \param input [., C, ...]
+                /// \param gamma gamma scaling for normalized value. [C]
+                /// \param beta bias added to the scaled normalized value [C]
+                /// \param mean value for mean normalization [C]
+                /// \param variance value for variance normalization [C]
+                /// \param epsilon Avoids divsion by 0 if input has 0 variance
+                BatchNormInference(const Output<Node>& input,
+                                   const Output<Node>& gamma,
+                                   const Output<Node>& beta,
+                                   const Output<Node>& mean,
+                                   const Output<Node>& variance,
+                                   double epsilon);
+
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                void validate_and_infer_types() override;
+
+                double get_eps_value() const { return m_epsilon; }
+                void set_eps_value(double epsilon) { m_epsilon = epsilon; }
+                std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+            private:
+                static constexpr size_t INPUT_DATA = 0;
+                static constexpr size_t INPUT_GAMMA = 1;
+                static constexpr size_t INPUT_BETA = 2;
+                static constexpr size_t INPUT_MEAN = 3;
+                static constexpr size_t INPUT_VARIANCE = 4;
```

Comment:
nit: May be move this to protected? So they can be used if a new version of the operation is introduced.

Reply:
We can do these fields to protected section if we face with situation, which requires it. (it will be backward compatible changes).

Reply:
BTW why do we need this variables if we have clear order of inputs in ctor?

Reply:
We also use it in validate and infer types. Also it allowed to use the code from opset1 BatchNormInference without any changes.